src/transaction.ts

```diff
@@ -402,6 +402,46 @@ class Transaction extends DatastoreRequest {
     this.save(entities);
   }
 
+  /**
+   * Merge the specified objects and add the merged object to the batch to be commited.
+   *
+   * @example
+   * const {Datastore} = require('@google-cloud/datastore');
+   * const datastore = new Datastore();
+   * const transaction = datastore.transaction();
+   *
+   * // Use an array, `excludeFromIndexes`, to exclude properties from indexing.
+   * // This will allow storing string values larger than 1500 bytes.
+   *
+   * const key = datastore.key(['Company', 123]);
+   * transaction.run((err) => {
+   *   if (err) {
+   *     // Error handling omitted.
+   *   }
+   *
+   *   transaction.get(key, (err, entity) => {
+   *     if (err) {
+   *       // Error handling omitted.
+   *     }
+   *
+   *     transaction.merge(entity, {meaningOfLife: 42});
+   *
+   *     transaction.commit((err) => {
+   *       if (!err) {
+   *         // Data saved successfully.
+   *       }
+   *     });
+   *   });
+   * });
+   */
+  merge(currentEntity: Entity, entityToMerge: Entity): void {
```

Comment:
I see that now we change this signature to allow for the flexibility of fetching entities beforehand in a separate transaction/operation and then providing them and using save because multiple save operations can be run at the same time without an error. But is the breaking change really worth it?

It will mean:
- Users need to change how they use merge
- They can currently work around this merge limitation by using `.save` anyway
- Not sure yet, but there may be a chance that just deleting local merging at https://github.com/googleapis/nodejs-datastore/blob/d854b57ed2b9b9f9ea148daedeb6d64e4d99de4d/src/request.ts#L956 gives us what we need

Reply:
I take back the third point. I hadn't realized `upsert` is different from a `merge` in that for a particular key, `upsert` replaces the current data entirely and merge merges the properties of the entity passed in with the entity currently saved.

I think we could add a new option for such merges to be handled by the backend and then we do not need a breaking change.